tests/js/spec/views/dashboardsV2/widgetBuilder/widgetBuilder.spec.tsx

```diff
@@ -1329,4 +1390,117 @@ describe('WidgetBuilder', function () {
       expect(mockModal).toHaveBeenCalled();
     });
   });
+
+  describe('group by field', function () {
+    it('does not contain functions as options', async function () {
+      renderTestComponent({
+        query: {displayType: 'line'},
+        orgFeatures: [...defaultOrgFeatures, 'new-widget-builder-experience-design'],
+      });
+
+      await screen.findByText('Group your results');
+
+      expect(screen.getByText('Select group')).toBeInTheDocument();
+
+      userEvent.click(screen.getByText('Select group'));
+
+      // Only one f(x) field set in the y-axis selector
+      expect(screen.getByText('f(x)')).toBeInTheDocument();
+    });
+
+    it('adds more fields when Add Group is clicked', async function () {
+      renderTestComponent({
+        query: {displayType: 'line'},
+        orgFeatures: [...defaultOrgFeatures, 'new-widget-builder-experience-design'],
+      });
+
+      await screen.findByText('Group your results');
+      userEvent.click(screen.getByText('Add Group'));
+      expect(await screen.findAllByText('Select group')).toHaveLength(2);
+    });
+
+    it('allows adding up to GROUP_BY_LIMIT fields', async function () {
+      renderTestComponent({
+        query: {displayType: 'line'},
+        orgFeatures: [...defaultOrgFeatures, 'new-widget-builder-experience-design'],
+      });
+
+      await screen.findByText('Group your results');
+
+      for (let i = 0; i < 19; i++) {
+        userEvent.click(screen.getByText('Add Group'));
+      }
+
+      expect(screen.queryByText('Add Group')).not.toBeInTheDocument();
+      expect(await screen.findAllByText('Select group')).toHaveLength(20);
+    });
+
+    it('allows deleting groups until there is one left', async function () {
+      renderTestComponent({
+        query: {displayType: 'line'},
+        orgFeatures: [...defaultOrgFeatures, 'new-widget-builder-experience-design'],
+      });
+
+      await screen.findByText('Group your results');
+      userEvent.click(screen.getByText('Add Group'));
+      expect(screen.getAllByLabelText('Remove group')).toHaveLength(2);
+
+      userEvent.click(screen.getAllByLabelText('Remove group')[1]);
+      expect(screen.queryByLabelText('Remove group')).not.toBeInTheDocument();
+    });
+
+    // it("doesn't reset group by when changing y-axis", async function () {
```

Comment:
These are triggering the `TypeError: Cannot read properties of null (reading 'createEvent')` error. I've commented them out for now for CI but we definitely need to look into this error. It's got something to do with fetching data because these would all trigger a request.

Reply:
oh these tests are annoying... we can check that and fix it in a separate PR